### PR TITLE
Add telemetry for gremlin queries

### PR DESCRIPTION
**Pull request by StephenWeatherford**
1) Refactors functions in utils.ts -> telemetry.ts, vscodeUtils.ts
2) Move a couple files to utils 
3) Pull telemetry code out of wrapAsyncCallback into telemetry.ts
4) Add telemetry for graph queries

### CODE REVIEW

File: src/utils/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+}
+
+export async function callWithTelemetry<T>(eventName: string, callback: (telemetryProperties: { [key: string]: string; }, measurements: { [key: string]: number }) => Promise<void>): Promise<void> {
+    const start = Date.now();
+    let properties: { [key: string]: string; } = {};
+    properties.result = 'Succeeded';
+    let measurements: { [key: string]: number; } = {};
+    let errorData: ErrorData | undefined = null;
+    let result: T = undefined;
```
Comment (EricJizbaMSFT) [inline src/utils/telemetry.ts]: I don't think result is ever used? And I think this is the only use of T? Could remove both

File: src/mongo/commands.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export class MongoCommands {
 				if (command.name === 'findOne') {
 					await editorManager.showDocument(new MongoFindOneResultEditor(database, command.collection, result));
 				} else {
-					await util.showNewFile(result, extensionPath, 'result', '.json', activeEditor.viewColumn + 1);
+					await vscodeUtil.showNewFile(result, extensionPath, 'result', '.json', activeEditor.viewColumn + 1);
```
Comment (PrashanthCorp) [inline src/mongo/commands.ts]: I expect a small merge conflict on this line. Just a heads-up.

Comment (PrashanthCorp) [inline src/mongo/commands.ts]: This block of code was refactored a bit. Heads-up about any merge conflicts.

File: src/utils/vscodeUtils.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+export function getOutputChannel(): vscode.OutputChannel {
+    return outputChannel;
+}
+
+export interface IDisposable {
+    dispose(): void;
+}
+
+export function dispose<T extends IDisposable>(disposables: T[]): T[] {
```
Comment (PrashanthCorp) [inline src/utils/vscodeUtils.ts]: I realize this is a file rename, so this line of code may not be down to you. 

This [documentation page](https://vshaxe.github.io/vscode-extern/vscode/Disposable.html) mentions a void return.  Any reason why we're returning an array here?

**Pull request merged**